docs/smooth-grid.md

```diff
@@ -0,0 +1,137 @@
+# Roblox Terrain Binary Format
+
+This document describes the Terrain binary format. In this format there is no field for Version, so it is assumed that any future changes will be additions to the existing format or a new format entirely. This specification does not include the adjacent PhysicsGrid binary format, only SmoothGrid.
+
+# Contents
+
+- [Document Conventions](#document-conventions)
+- [File Structure](#file-structure)
+- [Data Types](#data-types)
+  - [Chunk](#chunk)
+  - [Voxel](#voxel)
+  - [Voxel.Flag](#voxel.flag)
+  - [Material](#material)
+
+## Document Conventions
+
+This document assumes a basic understanding of Rust's convention for numeric types. For example:
+
+- `u32` is an unsigned 32-bit integer
+- `f32` is a 32-bit floating point number
+
+All numeric types are little endian. Floats are stored as a dividend of their type's numeric maximum; as such, a `f8` with a value of `0x31` translates to `0x31 / 0xFF`, or `~0.192`. Floats are unsigned.
+
+Unless otherwise noted, all structs in this document are assumed to be stored with their components in the sequence listed without any modification. That is, if a struct is listed as being composed of an `i32` and an `f32`, it can be assumed that it is stored as an `i32` followed by an `f32`.
+
+## File Structure
+
+The first two bytes of the blob are `0x01`, which is a magic number, followed by `0x05`, which is a logarithm base 2 of the chunk size in voxels. The default chunk size is 32<sup>3</sup> (32768).
+
+Size values other than `0x05` are normally not achievable with Studio or other tools, but values between `0x00` (inclusive) and `0x08` (inclusive) can still be deserialized by the engine, giving a theoretical range of possible chunk sizes between 1<sup>3</sup> and 256<sup>3</sup>. However, the engine splits chunks differeing in size from 32<sup>3</sup> back into the default chunk size. Any size value other than `0x05` has poorly tested and undefined behavior.
+
+Immediately following the header is an array of chunks, each of which must contain enough voxels to reach the maximum count (32<sup>3</sup>). Chunks are ascendingly ordered by X, then Y, then Z based on their position in the world. Each chunk represents a cube of 128<sup>3</sup> units in world space.
```

Comment:
I think we went over this on Discord, but chunk sizes aren't fixed and are in fact `(2^CHUNK_SIZE)^3`, correct? If this is the case then this should be adjusted to indicate that somehow.

Reply:
yes its dynamic
i thought i got this bit when I resolved it but alas I did not